src_c/simd_blitters_avx2.c

```diff
@@ -928,3 +928,174 @@ blit_blend_rgb_min_avx2(SDL_BlitInfo *info)
+            /* one pixel at a time - same as current sse2 version */
+            LOOP_UNROLLED4(
+                {
+                    Uint32 alpha = *srcp & amask;
+                    if (alpha == 0) {
+                        /* do nothing */
+                    }
+                    else if (alpha == amask) {
+                        *dstp = *srcp;
+                    }
+                    else {
+                        mm_src = _mm_cvtsi32_si128(*srcp);
+                        /*mm_src = 0x000000000000000000000000AARRGGBB*/
+                        mm_src = _mm_unpacklo_epi8(mm_src, mm_zero);
+                        /*mm_src = 0x000000000000000000AA00RR00GG00BB*/
+                        mm_dst = _mm_cvtsi32_si128(*dstp);
+                        /*mm_dst = 0x000000000000000000000000AARRGGBB*/
+                        mm_dst = _mm_unpacklo_epi8(mm_dst, mm_zero);
+                        /*mm_dst = 0x000000000000000000AA00RR00GG00BB*/
+
+                        mm_alpha = _mm_cvtsi32_si128(alpha);
+                        /* alpha -> mm_alpha (000000000000A000) */
+                        mm_alpha = _mm_srli_si128(mm_alpha, 3);
+                        /* mm_alpha >> ashift -> mm_alpha(000000000000000A) */
+                        mm_alpha = _mm_unpacklo_epi16(mm_alpha, mm_alpha);
+                        /* 0000000000000A0A -> mm_alpha */
+                        mm_alpha = _mm_unpacklo_epi32(mm_alpha, mm_alpha);
+                        /* 000000000A0A0A0A -> mm_alpha2 */
+
+                        /* pre-multiplied alpha blend */
+                        mm_sub_dst = _mm_add_epi16(mm_dst, mm_ones);
+                        mm_sub_dst = _mm_mullo_epi16(mm_sub_dst, mm_alpha);
+                        mm_sub_dst = _mm_srli_epi16(mm_sub_dst, 8);
+                        mm_dst = _mm_add_epi16(mm_src, mm_dst);
+                        mm_dst = _mm_sub_epi16(mm_dst, mm_sub_dst);
+                        mm_dst = _mm_packus_epi16(mm_dst, mm_zero);
+
+                        *dstp = _mm_cvtsi128_si32(mm_dst);
+                    }
+
+                    srcp += srcpxskip;
+                    dstp += dstpxskip;
+                },
+                n, pre_8_width);
+        }
+        srcp256 = (__m256i *)srcp;
+        dstp256 = (__m256i *)dstp;
+        if (post_8_width > 0) {
+            /*8 pixels at a time, need to use shuffle to get everything
+                lined up - see mul for an example*/
+            LOOP_UNROLLED4(
+                {
+                    mm256_src = _mm256_loadu_si256(srcp256);
+                    mm256_dst = _mm256_loadu_si256(dstp256);
+
+                    /* insert 8 pixel at a time blend here */
+
+                    /* do everything A set first */
+                    mm256_dstA =
+                        _mm256_shuffle_epi8(mm256_dst, mm256_shuff_mask_A);
+                    mm256_src_shuff =
+                        _mm256_shuffle_epi8(mm256_src, mm256_shuff_mask_A);
```

Comment:
I would appreciate some additional comments in the algorithm, especially for what state the shuffles put things into, since they are hard to decode.